src/models/neural_networks/base.py

```diff
@@ -322,8 +312,34 @@ def _get_background(self, sample_size: int = 150) -> List[torch.Tensor]:
             torch.stack(output_static),
         ]
 
-    def _one_hot_months(self, indices: torch.Tensor) -> torch.Tensor:
-        return torch.eye(14, device=self.device)[indices.long()][:, 1:-1]
+    def _one_hot(self, indices: torch.Tensor, num_vals: int) -> torch.Tensor:
+        if len(indices.shape) > 1:
+            indices = indices.squeeze(-1)
+        return torch.eye(num_vals + 2, device=self.device)[indices.long()][:, 1:-1]
+
+    def _input_to_tuple(
+        self, x: Union[Tuple[torch.Tensor, ...], TrainData]
+    ) -> Tuple[torch.Tensor, ...]:
+        # mypy totally fails to handle what's going on here
+
+        if type(x) is TrainData:  # type: ignore
+            return (  # type: ignore
+                x.historical,  # type: ignore
+                self._one_hot(x.pred_months, 12),  # type: ignore
+                x.latlons,  # type: ignore
+                x.current,  # type: ignore
+                x.yearly_aggs,  # type: ignore
+                self._one_hot(x.static, self.num_locations) if self.static == "embeddings" else x.static,  # type: ignore
+            )
+        else:
```

Comment:
in what situations is it not `TrainData`? When it's testing so it's a `torch.Tensor`

Reply:
Yea, if the dataloader is returning training data, its a list of tensors.

Also, if the batch is chunked and shuffled, that will also return a list of tensors.